datafusion-proto/src/from_proto.rs

```diff
@@ -992,32 +1017,86 @@ impl TryFrom<&protobuf::LogicalExprNode> for Expr {
-                    // ScalarFunction::Array => Ok(array((&args[0]).try_into()?)),
-                    // // ScalarFunction::Nullif => Ok(nulli((&args[0]).try_into()?)),
-                    ScalarFunction::Datepart => {
+                    ScalarFunction::DatePart => {
                         Ok(date_part((&args[0]).try_into()?, (&args[1]).try_into()?))
                     }
-                    ScalarFunction::Datetrunc => {
+                    ScalarFunction::DateTrunc => {
                         Ok(date_trunc((&args[0]).try_into()?, (&args[1]).try_into()?))
                     }
-                    // ScalarFunction::Md5 => Ok(md5((&args[0]).try_into()?)),
                     ScalarFunction::Sha224 => Ok(sha224((&args[0]).try_into()?)),
                     ScalarFunction::Sha256 => Ok(sha256((&args[0]).try_into()?)),
                     ScalarFunction::Sha384 => Ok(sha384((&args[0]).try_into()?)),
                     ScalarFunction::Sha512 => Ok(sha512((&args[0]).try_into()?)),
                     ScalarFunction::Digest => {
                         Ok(digest((&args[0]).try_into()?, (&args[1]).try_into()?))
                     }
+                    //ScalarFunction::ToTimestampMillis => Ok(to_tome)
+                    ScalarFunction::Ascii => Ok(ascii((&args[0]).try_into()?)),
+                    ScalarFunction::BitLength => Ok((&args[0]).try_into()?),
+                    ScalarFunction::CharacterLength => {
+                        Ok(character_length((&args[0]).try_into()?))
+                    }
+                    ScalarFunction::Chr => Ok(chr((&args[0]).try_into()?)),
+                    // ScalarFunction::ConcatWithSeparator => {
+                    //     Ok(concat_ws((&args[0]).try_into()?, (&args[1]).try_into()?))
+                    // }
+                    ScalarFunction::InitCap => Ok(ascii((&args[0]).try_into()?)),
+                    ScalarFunction::Left => {
+                        Ok(left((&args[0]).try_into()?, (&args[1]).try_into()?))
+                    }
+                    ScalarFunction::Random => Ok(random()),
+                    ScalarFunction::Repeat => {
+                        Ok(repeat((&args[0]).try_into()?, (&args[1]).try_into()?))
+                    }
+                    ScalarFunction::Replace => Ok(replace(
+                        (&args[0]).try_into()?,
+                        (&args[1]).try_into()?,
+                        (&args[2]).try_into()?,
+                    )),
+                    ScalarFunction::Reverse => Ok(reverse((&args[0]).try_into()?)),
+                    ScalarFunction::Right => {
+                        Ok(right((&args[0]).try_into()?, (&args[1]).try_into()?))
+                    }
+                    // ScalarFunction::Rpad => Ok(rpad(vec![])),
+                    // ScalarFunction::RegexpReplace => {
+                    //     Ok(regexp_replace(vec![]))
+                    // }
```

Comment:
Add  issue #2009

Reply:
Can we fix the #2009 and #2010  in this pull request?

Reply:
I agree that we should implement these functions in this pr as much as possible. Only leave the controversial ones.

Reply:
Sure, will find a path how to fix these.

Reply:
Please let me know if it's ready for the next round of review.

Reply:
@yjshen PTAL